pkg/database/realm.go

```diff
@@ -716,6 +716,18 @@ func (db *Database) CreateRealm(name string) (*Realm, error) {
 	return realm, nil
 }
 
+func (db *Database) FindRealmByRegion(region string) (*Realm, error) {
```

Comment:
I've moved us<sup>1</sup> away from database lookup functions returning nil because it tends to lead to NPEs. I'd like this to return an error instead, and then the controller should use `database.IsNotFound` to flow control.

<sup>1. </sup> The exception is when the function returns a list. In that case, the empty list (but NOT an uninitialized list) is returned.